### ISSUE TITLE
Don't auto-hide on losing input focus (full-screen)

### ISSUE THREAD

**Issue by dsalt** [00-Evan/shattered-pixel-dungeon#1870]
I find that auto-hiding the window is *less than ideal* with multiple monitors, particularly without pointer grab. Either the window should remain visible regardless or this behaviour should be configurable (“borderless window”).

**Comment by 00-Evan** [comment]
Which operating system are you using? There is no explicit behavior on Shattered's part that causes it to do this, but it may be a consequence of using fullscreen. Unfortunately borderless windowed functionality causes a host of issues on macOS and Linux, which is why the game doesn't use it.

**Comment by dsalt** [comment]
Devuan stable, Xfce4 desktop.

I can simulate borderless window mode by using windowed mode and xfwm4's own full-screen toggle. I can also have devilspie2 force this kind of full-screen mode (so long as the window title remains stable, at least on startup). But I think it better that it's provided by the game itself.

I don't think that I've ever seen problems caused by this. It seems like it could be something which is specific to some window manager (or some types of WM).

**Comment by dsalt** [comment]
GNOME? That environment having annoying behaviour doesn't surprise me…

(You _could_ implement and add a warning, e.g. “may not function as intended in some desktop environments”. But at least it works here, if only via external means.)

**Comment by dsalt** [comment]
Were you *maximising* the window? That's not the same thing… It occurred to me to test that instead of full-screen on my secondary monitor (which has panels displayed, some of which prevent maximisation from filling the display): I see behaviour which matches your description.

As I have libGDX source here (for that backslash problem), I decided to have a look to see what it does regarding full-screen etc., and I see a function which does what I want here. I've successfully tested fullscreen mode for the desired behaviour with the following patch:
```diff
diff --git a/desktop/src/main/java/com/shatteredpixel/shatteredpixeldungeon/desktop/DesktopLauncher.java b/desktop/src/main/java/com/shatteredpixel/shatteredpixeldungeon/desktop/DesktopLauncher.java
index 0dee850d1..912b8c7b6 100644
--- a/desktop/src/main/java/com/shatteredpixel/shatteredpixeldungeon/desktop/DesktopLauncher.java
+++ b/desktop/src/main/java/com/shatteredpixel/shatteredpixeldungeon/desktop/DesktopLauncher.java
@@ -178,6 +178,7 @@ public class DesktopLauncher {
                //going fullscreen on launch is a bit buggy
                // so game always starts windowed and then switches in DesktopPlatformSupport.updateSystemUI
                //config.setFullscreenMode(Lwjgl3ApplicationConfiguration.getDisplayMode());
+               config.setAutoIconify(false);
 
                //records whether window is maximized or not for settings
                DesktopWindowListener listener = new DesktopWindowListener();
```

**Comment by 00-Evan** [comment]
Borderless windowed is maximizing the window without window decorations.

Afraid that change to autoIconify causes fullscreen mode to completely break in other contexts. The fullscreen window will never minimize, making it impossible to alt-tab outside of the fullscreen view in windows, for example.

Also, I appreciate the intent to help but I do not accept code contributions, both because Shattered is a solo project and due to licensing concerns, if you send me suggestion code it essentially prevents me from using it.

**Comment by dsalt** [comment]
> Borderless windowed is maximizing the window without window decorations.

In appearance, it is close to that. However, I think that the name commonly used for it is misleading: I don't think that I've seen any actual implementation which is anything other than full-screen without the minimisation (iconisation) on focus loss.

(Some other games don't call it “borderless windowed” – they call it “full-screen” but may also provide an option, typically called “exclusive mode”, which when on enables minimisation on focus loss. SPD's full-screen mode is, in that terminology, full-screen exclusive mode.)

Regarding that patch: it exists to illustrate the intended behaviour in the most minimal way possible, to say “calling this method does the job”; it specifically is *not* a proper implementation; given your licensing concerns, I chose *not* to create a patch which could reasonably be used in production (for one thing, there'd be configuration UI stuff in it somewhere). Regardless, you know which method does the job now, and I intended nothing more than that (and if there's a licensing concern *there*, something's badly broken). (Should I have just named the method?)
